File: samples/cs/03sqrt/sqrt.cs
using System;

namespace SquareRoot {
    class Exec {

        static double FindSqrt(double number) {
            double x = number;
            double y = 1.0;
            const double e = 0.0000001;
            while ((x - y) > e) {
                x = (x + y) / 2;
                y = number / x;
            }
            return x;
        }

        static void Main(string[] args) {
            if (args.Length < 1) {
                Console.Error.WriteLine("Error: Expected to receive at least 1 argument ./sqrt <number>  [<numbe> ...].");
                System.Environment.Exit(1);
            }
            foreach (string arg in args) {
                double number = 0;
                try {
                    number = Double.Parse(arg);
                } catch (System.FormatException) {
                    Console.Error.WriteLine(string.Format("Error: Expected to receive 1 argument that was a number. Received {0}", arg));
                    System.Environment.Exit(2);
                }
                Console.WriteLine(number.ToString() + " => " + FindSqrt(number).ToString());
            }
        }
    }
}

File: samples/js/03sqrt/sqrt.js
function findSqrt(number) {
    let x = number; 
    let y = 1; 
    let e = 0.000000001; /* e decides the accuracy level*/
    while (x - y > e) { 
        x = (x + y) / 2; 
        y = number / x; 
    } 
    return x; 
}


if (require.main === module) {
    if (process.argv.length < 3) {
        console.error("Error: Usage node sqrt.js <number> [<number> ...]");
        process.exit(1);
    }
    process.argv.slice(2).map(arg=>{
        const [number] = arg;
        const _number = Number(number);
        if (isNaN(_number)) {
            console.error(`Error: <number> must be a numeric value. Received '${number}'`);
            process.exit(2);
        }
        console.log(`${number} => ${findSqrt(_number)}`);
    });
}
